initrd/directory.go

```diff
@@ -77,6 +77,10 @@ func (initrd *directory) Build(ctx context.Context) (string, error) {
 		return "", fmt.Errorf("could not create output directory: %w", err)
 	}
 
+	if initrd.opts.fsType == FsTypeErofs {
+		return initrd.opts.output, convertToErofs(initrd.opts.output, initrd.path, false, !initrd.opts.keepOwners)
+	}
+
```

Comment:
This needs some thinking as it feels a bit of a hack to add this before a large chunk of code which does this for a different file system type. It's better to consolidate the two different file system types into interfaces or something and homogeneously `switch` between them depending on input format.